### PR TITLE
Create reconnect button/command

### PR DESCRIPTION
**Pull request by DusterTheFirst**
Created a reconnect button and command for easy resuming of Discord RPC without refreshing the window

*(#28 but not broken)*

### CODE REVIEW

**Comment by DusterTheFirst** [comment]
image

File: src/extension.ts [hunk 1]
```diff
@@ -10,7 +10,9 @@ import {
 	ExtensionContext,
 	window,
 	workspace,
-	WorkspaceFolder
+	WorkspaceFolder,
+	StatusBarItem,
```
Comment (iCrawl) [inline src/extension.ts]: This should be ordered from A-Z.

File: src/extension.ts [hunk 2]
```diff
@@ -58,8 +62,18 @@ export function activate(context: ExtensionContext) {
 		window.showInformationMessage('Disabled Discord Rich Presence for this workspace.');
 	});
 
+	// Register the `discord.reconnect` command
+	const reconnecter = commands.registerCommand('discord.reconnect', async () => {
+		if (rpc) try { await destroyRPC(); } catch {}
+		initRPC(config.get('clientID'), true);
+		window.showInformationMessage('Reconnecting to Discord RPC');
+
+		if (statusBarIcon)
```
Comment (iCrawl) [inline src/extension.ts]: Not liking how you newline the if-statement here. This is def. a style change and should not happen, see 5 lines above how you inline it usually when its one statement.

File: src/extension.ts [hunk 3]
```diff
@@ -69,12 +83,21 @@ export async function deactivate() {
...
 
 // Initalize the RPC systems.
-function initRPC(clientID: string): void {
+function initRPC(clientID: string, loud?: boolean): void {
 	// Update the RPC variable with a new RPC Client.
 	rpc = new Client({ transport: 'ipc' });
 
 	// Once the RPC Client is ready, set the activity.
 	rpc.once('ready', () => {
+		if (loud) {
```
Comment (iCrawl) [inline src/extension.ts]: This should be one-lined without curlies.

File: src/extension.ts [hunk 4]
```diff
@@ -69,12 +83,21 @@ export async function deactivate() {
...
+			window.showInformationMessage('Successfully reconnected to Discord RPC');
+		}
+
+		// Remove icon if connected
+		if (statusBarIcon) {
```
Comment (iCrawl) [inline src/extension.ts]: Same, also one-lined.

File: src/extension.ts [hunk 5]
```diff
@@ -117,17 +140,32 @@ function initRPC(clientID: string): void {
 	rpc.login(clientID).catch(async error => {
 		if (reconnectTimer) {
 			// Destroy and dispose of everything after a default of 20 reconnect attempts
-			if (reconnectCounter >= config.get('reconnectThreshold')) await destroyRPC();
-			else return;
+			if (reconnectCounter >= config.get('reconnectThreshold')) {
+				createButon();
+				await destroyRPC();
+			} else return;
```
Comment (iCrawl) [inline src/extension.ts]: If you use curlies for the if also use it for the else like you do in the createButton function below.

File: src/extension.ts [hunk 6]
```diff
@@ -117,17 +140,32 @@ function initRPC(clientID: string): void {
...
 		if (!config.get('silent')) {
 			if (error.message.includes('ENOENT')) window.showErrorMessage('No Discord Client detected!');
 			else window.showErrorMessage(`Couldn't connect to Discord via RPC: ${error.message}`);
+			createButon();
 		}
 	});
 }
 
-// Cleanly destroy the RPC client (if it isn't already).
+/** Create reconnect button */
```
Comment (iCrawl) [inline src/extension.ts]: Also this should probably be a normal // comment instead of /** */

**Comment by DusterTheFirst** [comment]
Sorry about those, not used to following a style. If you look at my work, it is a mess, I should probably enforce a style like this

**Comment by iCrawl** [comment]
No worries.

**Pull request merged**